src/producer/mod.rs

```diff
@@ -106,6 +107,32 @@ impl<S: Read + Write> Producer<S> {
         self.c.issue(&Push::from(job))?.await_ok()
     }
 
+    /// Enqueue numerous jobs on the Faktory server.
+    ///
+    /// Provided you have numerous jobs to submit, using this method will be more efficient as compared
+    /// to calling [`enqueue`](Producer::enqueue) multiple times.
+    ///
+    /// The returned `Ok` result will contain a tuple of enqueued jobs count and an option of a hash map
+    /// with job ids mapped onto error messages. Therefore `Ok(n, None)` will indicate that all n jobs
+    /// have been enqueued without errors.
+    ///
+    /// Note that this is not an all-or-nothing operation: jobs that contain errors will not be enqueued,
+    /// while those that are error-free _will_ be enqueued by the Faktory server.
+    pub fn enqueue_many(
+        &mut self,
+        jobs: &[Job],
+    ) -> Result<(usize, Option<HashMap<String, String>>), Error> {
```

Comment:
What do you think of doing something like this?
```suggestion
    pub fn enqueue_many<J>(
        &mut self,
        jobs: J,
    ) -> Result<(usize, Option<HashMap<String, String>>), Error>
      where J: IntoIterator<Item = Job>,
            J::IntoIter: ExactSizeIterator
    {
        let jobs = jobs.into_iter();
        let njobs = jobs.len();
```
You'd also want to update `PushBulk` accordingly (although it'll only need an `Iterator<Item = Job>` bound).

Reply:
Please see the updated version, as for the `PushBulk` - updating it expect a vector of jobs. I guess I might be missing a way to collect an iterator behind a shared refence in `issue`, while `Serialize` is not automatically implemented for the `Iterator<Item = S> where S: Serialize`